source/event.c

```diff
@@ -43,6 +43,7 @@ static NS_LIST_DEFINE(free_event_entry, arm_core_event_s, link);
 // Statically allocate initial pool of events.
 #define STARTUP_EVENT_POOL_SIZE 10
 static arm_core_event_s startup_event_pool[STARTUP_EVENT_POOL_SIZE];
+static uint16_t event_list_cur_size = 0;
```

Comment:
Why `uint16_t`?

Technically it does not same RAM because of alignment things.
It causes more code to be generated because all registers are 32bits in ARM so operating 16bit requires extra operations.

Reply:
Shuold I set int?

Reply:
Actaully linked list also return uint16_t  return vaue

Reply:
nanomesh4-rel is used at msp430 so uint16_t should not be a problem.